conan/tools/env/environment.py

```diff
@@ -251,16 +251,30 @@ def __getitem__(self, name):
     def keys(self):
         return self._values.keys()
 
-    def get(self, name, default=None):
+    def get(self, name, default=None, placeholder=None):
+        """
+        if placeholder is specified it means that we want to get a placeholder instead of the
+        real value of the env-var
```

Comment:
The name `placeholder` and this comment confused me a little bit. Am I understand this correctly, in that `placeholder` only makes sense for variables that have had values set with `append` or `prepend` rather than with `define`, and that the placeholder applies to the pre-existing value? So rather than resolving that value, a placeholder is referenced instead, where `{name}` in the placeholder will resolve to the variable name. 

I'm not sure if `placeholder` as a name captures the functionality, and from this comment I understood something different - maybe something like "if placeholder is specified, it will be used in place of the pre-existing value of the env-var, where {name} will resolve to the variable name"?

Reply:
Yes, only ``append`` and ``prepend`` will contain an internal list like ``["userdata1", CURRENT_ENV_VAR_VALUE, "userdata2]``, because ``define()`` will overwrite existing env-var value.

The idea is that when ``placeholder`` is defined, instead of replacing with the current env-var value, it is replaced with a typical placeholder, like ``%envvar%``, ``$envvar`` as string literal.

I see what you mean, but failed to bring a better name, any suggestion? Or just improving the docstring?

Reply:
I like the word "reference", since we're dealing with how whether a variable is resolved to a reference - 
e.g. something along the lines of:

`variable_reference`: if specified, use a variable reference instead of the pre-existing value of environment variable, where `{name}` can be used to refer to the name of the variable. 

I think this naming would be consistent with https://www.gnu.org/software/make/manual/html_node/Reference.html

Reply:
Sounds good, I like it more, I'll change it

Reply:
Done